clap_man/tests/generate.rs

```diff
@@ -27,3 +27,73 @@ fn render_manpage() {
 
     Man::new(app).render(&mut io::sink()).unwrap();
 }
+
+#[test]
+fn verify_argument_ordering_is_preserved() {
+    let app = App::new("prog")
+        .arg(
+            Arg::new("a") // Typically args are grouped alphabetically by name.
+                // Args without a display_order have a value of 999 and are
+                // displayed alphabetically with all other 999 valued args.
+                .long("last")
+                .short('o')
+                .takes_value(true)
+                .help("Some help and text: Should be last"),
+        )
+        .arg(
+            Arg::new("e")
+                .long("fourth")
+                .short('R')
+                .takes_value(true)
+                .display_order(4)
+                .help("I should be fourth!"),
+        )
+        .arg(
+            Arg::new("d")
+                .long("third")
+                .short('Q')
+                .takes_value(true)
+                .display_order(3)
+                .help("I should be third!"),
+        )
+        .arg(
+            Arg::new("c")
+                .long("second")
+                .short('P')
+                .takes_value(true)
+                .display_order(2)
+                .help("I should be second!"),
+        )
+        .arg(
+            Arg::new("b")
+                .long("first")
+                .short('O')
+                .takes_value(true)
+                .display_order(1) // In order to force this arg to appear *first*
+                // all we have to do is give it a value lower than 999.
+                // Any other args with a value of 1 will be displayed
+                // alphabetically with this one...then 2 values, then 3, etc.
+                .help("I should be first!"),
+        );
+
+    let mut buf = Vec::new();
+    Man::new(app).render(&mut buf).unwrap();
+    let s = match std::str::from_utf8(&buf) {
+        Ok(v) => v,
+        Err(e) => panic!("Invalid UTF-8 sequence: {}", e),
+    };
```

Comment:
Side note: I have a snapshot testing crate that could make life easier with these with some small changes.